indexer/validate_and_format_contacts.cpp

```diff
@@ -406,4 +416,97 @@ bool ValidateLinePage(string const & page)
   return (domain == "line.me" || strings::EndsWith(domain, ".line.me"));
 }
 
+bool isSocialContactTag(string_view tag)
+{
+  return tag == "contact:instagram" || tag == "contact:facebook" || tag == "contact:twitter"
+         || tag == "contact:vk" || tag == "contact:line";
+}
+
+bool isSocialContactTag(MapObject::MetadataID const metaID)
+{
+  return metaID == MapObject::MetadataID::FMD_CONTACT_INSTAGRAM ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_FACEBOOK ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_TWITTER ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_VK ||
+         metaID == MapObject::MetadataID::FMD_CONTACT_LINE;
+}
+
+// Functions ValidateAndFormat_{facebook,instagram,twitter,vk}(...) by default strip domain name
+// from OSM data and user input. This function prepends domain name to generate full URL.
+string socialContactToURL(string_view tag, string_view value)
+{
+  if(value.empty())
+    return {};
+
+  if (tag == "contact:instagram")
+    return "https://instagram.com/" + string(value);
+  if (tag == "contact:facebook")
+    return "https://facebook.com/" + string(value);
+  if (tag == "contact:twitter")
+    return "https://twitter.com/" + string(value);
+  if (tag == "contact:vk")
+    return "https://vk.com/" + string(value);
+
+  if (tag == "contact:line")
+  {
+    if (value.find('/') == string::npos) // 'value' is a username.
+      return "https://line.me/R/ti/p/@" + string(value);
+    else // 'value' is an URL.
+      return "https://" + string(value);
+  }
+
+  return string(value);
```

Comment:
{} is a safer constructor version than () here and above.
```suggestion
  return string{value};
```

Reply:
Fixed.